example/eblob_backend.c

```diff
@@ -86,15 +86,28 @@ static int eblob_read_params_compare(const void *p1, const void *p2)
 }
 
 /* Pre-callback that formats arguments and calls ictl->callback */
-static int blob_iterate_callback_common(struct eblob_disk_control *dc, void *data, void *priv, int no_meta) {
+static int blob_iterate_callback_common(struct eblob_disk_control *dc, int fd, uint64_t data_offset, void *priv, int no_meta) {
 	struct dnet_iterator_ctl *ictl = priv;
 	struct dnet_ext_list elist;
 	struct eblob_backend_config *c = ictl->iterate_private;
 	uint64_t size;
+	void *data, *saved_data;
 	int err;
 
 	assert(dc != NULL);
-	assert(data != NULL);
+
+	size = dc->disk_size;
+	saved_data = data = malloc(size);
```

Comment:
From library/dnet.c: dnet_iterator_callback_common:
position = combined = malloc(size);

But now I see that it allocates memory only when DNET_IFLAGS_DATA.
So it is possible to pass fd to this function and avoid extra memcpy.

In current version it (dnet_iterator_callback_common) may allocate memory enough to handle whole file data.